dot/state/epoch.go

```diff
@@ -380,10 +381,10 @@ func (s *EpochState) getConfigDataFromDatabase(epoch uint64) (*types.ConfigData,
 	return info, nil
 }
 
-func retrieveFromMemory[T types.NextEpochData | types.NextConfigData](nextEpochMap map[uint64]map[common.Hash]T,
-	blockState *BlockState, epoch uint64, header *types.Header) (*T, error) {
+type nextEpochMap[T types.NextEpochData | types.NextConfigData] map[uint64]map[common.Hash]T
 
-	atEpoch, has := nextEpochMap[epoch]
+func (nem nextEpochMap[T]) Retrieve(blockState *BlockState, epoch uint64, header *types.Header) (*T, error) {
```

Comment:
could it be not exported?

Reply:
well the type is private, but it seems like there should be an exported method on any type IMO.